File: LeetCode75-sols/Two_Pointers/Java/Is_Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int s_ptr = 0;
        // return true if s is empty string
        if (s.length() == 0) {
            return true;
        }
        
        for (int t_ptr = 0; t_ptr < t.length(); t_ptr++) {
            if (s_ptr > (s.length() - 1)) {
                break;
            }
            else if (s.charAt(s_ptr) == t.charAt(t_ptr)) {
                s_ptr++;
            }
        }
        if (s_ptr == s.length()) {
            return true;
        }
        else {
            return false;
        }
    }
}

File: LeetCode75-sols/Two_Pointers/CPP/Is_Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int s_ptr = 0;
        // Return True if s is empty string
        if (s.size() == 0) {
            return true;
        }

        for (int t_ptr = 0; t_ptr < t.size(); t_ptr++) {
            if (s_ptr > (s.size() - 1)) {
                break;
            }
            else if (s[s_ptr] == t[t_ptr]) {
                s_ptr++;
            }
        }
        if (s_ptr == s.size()) {
            return true;
        }
        else {
            return false;
        }
    }
};